app/bundles/CoreBundle/Resources/views/Modal/search_commands.html.twig

```diff
@@ -0,0 +1,417 @@
+<!-- Search commands modal -->
```

Comment:
How about breaking it into smaller files for readability and maintainability?

Reply:
That's something to consider
I initially placed HTML comments before each section to help these things to be found (e.g. a workflow where first you inspect that section using your browser dev tools to check the comment, then using a search functionality in your IDE to find that comment)

Perhaps separating in different files would increase complexity... In one of my PRs, I add a twig template for accordions that tend to work well for simpler things (or medium complexity with Twig % set % variables), but these modals have a lot of `<code>`, `<p>`, `translation strings` etc ordered in a specific way to ensure each thing is rendered without visual bugs that could appear in future changes to the template

I mean, stability is crucial, maybe costing just a bit of the maintainability

Idk, what do you think having this new info?

Reply:
I understand your point. However, I don't have a robust way of doing this at the moment. Let's park this readability for future

Reply:
This is really not a way to go. Bundles should have their own logic in its directory to keep the modular architecture. Most importantly, plugins cannot add their own search commands to this at all.

Let's find a better solution that won't break to the modular architecture and be extensible. I noticed that bundles are defining their search commands like this:

https://github.com/mautic/mautic/blob/5.x/app/bundles/LeadBundle/EventListener/SearchSubscriber.php#L118

The event that collects those search commands is dispatched only in this one place:

https://github.com/mautic/mautic/blob/b5d174f2d2b8582bc18101ddf09ed2559c34c604/app/bundles/CoreBundle/Controller/AjaxController.php#L188

So it can be moved to its own service so it can be used also to generate this list of available search commands. This would also ensure that the documentation is always up to date with implemented search commands.

Reply:
@escopecz  What exactly do you want me to do?

Reply:
With #13939 (this PR), I'm replacing what's being removed here:
- https://github.com/mautic/mautic/pull/13950

Reply:
@andersonjeccel it's hard to describe code. It is better to just write it. Here is what I meant:

https://github.com/mautic/mautic/compare/5.x...escopecz:search-commands-generated?expand=1

It has your changes plus this commit: https://github.com/mautic/mautic/commit/96a099bc56a7edbd9e0026bf37ed4229cc447d6c

And it let you get all the available commands that are being already generated by the subscribers in the twig file. Here is the structure:
![Screenshot 2024-08-30 at 10 26 05](https://github.com/user-attachments/assets/726771f5-a067-4a37-81e0-94ed2230363c)

Do you think you would be able to generate such a nice help content from this list of commands?

This way the content will be dynamic. If a command would be added, it would be also documented. It is also extensible by any plugin. And no code specific to 1 bundle will be in the CoreBundle.

Reply:
@escopecz 
It seems possible, but the output isn't matching all available commands...

API for example, that's what we have documented (in devdocs):

![image](https://github.com/user-attachments/assets/8bfbac69-26ae-4cd6-a842-05da27a42b08)

But only ID is showing from the JSON output of your code:

![image](https://github.com/user-attachments/assets/f8e3622f-e352-4bd1-9659-3ec6c1b614ca)

Any clue re. what's happening?
_(also seems to be missing all commands related to dynamic content, but they're not specific)_

Reply:
That's the beauty of this solution. OAuth 1.0a is no longer supported and I cannot find any support for the `redirecturi` in the code and it doesn't work when I test it in the UI either. The generated list is up to date. The docs may be outdated as they are not connected to the search command implementation. We should use the autogeneration of the docs from code to avoid this.

Reply:
Ok, but what about missing commands (dynamic content) or others that I might not be seeing yet?

Reply:
There is no search support for Dynamic Content. So I guess all the search commands that work will work also for all other entities.
![Screenshot 2024-09-04 at 13 46 31](https://github.com/user-attachments/assets/d47c2846-c4d8-4152-bc07-d4e711427bde)
If you look at the search docs for Dynamic Content, I think they were copied from Landing Pages as there is the `is:prefcenter` search command which is a thing only for landing pages.

Reply:
So these defaults should work for all  (dynamic content, focus items, marketing messages, SMS)?

![image](https://github.com/user-attachments/assets/c938c4b6-a8f9-4a1b-92de-eb7a1504f441)

Reply:
Looks like it.

Reply:
Which translation strings do you mean?

Reply:
I get it now. Yes, it seems that it would work even translated. When looking at https://github.com/mautic/mautic/blob/5.x/app/bundles/LeadBundle/Entity/LeadRepository.php#L721-L724 it checks the localized translation as well as the English version when processing the search command.